samples/app-sso/csharp/app_sso/Bots/TeamsMessagingExtensionsSearchAuthConfigBot.cs

```diff
@@ -0,0 +1,482 @@
+                        },
+                    },
+                };
+            }
+
+            var client = new SimpleGraphClient(tokenResponse.Token);
+            var profile = await client.GetMyProfile();
+            var heroCard = new ThumbnailCard
+            {
+                Title = "Thumbnail Card",
+                Text = $"Hello, {profile.DisplayName}",
+                Images = new List<CardImage> { new CardImage("https://raw.githubusercontent.com/microsoft/botframework-sdk/master/icon.png") },
+            };
+            var attachments = new MessagingExtensionAttachment(HeroCard.ContentType, null, heroCard);
+            var result = new MessagingExtensionResult("list", "result", new[] { attachments });
+            return new MessagingExtensionResponse(result);
+        }
+
+        protected override async Task<MessagingExtensionResponse> OnTeamsMessagingExtensionConfigurationQuerySettingUrlAsync(ITurnContext<IInvokeActivity> turnContext, MessagingExtensionQuery query, CancellationToken cancellationToken)
+        {
+            // The user has requested the Messaging Extension Configuration page.  
+            var escapedSettings = string.Empty;
+            var userConfigSettings = await _userConfigProperty.GetAsync(turnContext, () => string.Empty);
+            if (!string.IsNullOrEmpty(userConfigSettings))
+            {
+                escapedSettings = Uri.EscapeDataString(userConfigSettings);
+            }
+            return new MessagingExtensionResponse
+            {
+                ComposeExtension = new MessagingExtensionResult
+                {
+                    Type = "config",
+                    SuggestedActions = new MessagingExtensionSuggestedAction
+                    {
+                        Actions = new List<CardAction>
+                        {
+                            new CardAction
+                            {
+                                Type = ActionTypes.OpenUrl,
+                                Value = $"{_siteUrl}/searchSettings.html?settings={escapedSettings}",
+                            },
+                        },
+                    },
+                },
+            };
+        }
+
+        protected override async Task OnTeamsMessagingExtensionConfigurationSettingAsync(ITurnContext<IInvokeActivity> turnContext, JObject settings, CancellationToken cancellationToken)
+        {
+            // When the user submits the settings page, this event is fired.
+            if (settings["state"] != null)
+            {
+                var userConfigSettings = settings["state"].ToString();
+                await _userConfigProperty.SetAsync(turnContext, userConfigSettings, cancellationToken);
+            }
+        }
+
+        protected override async Task<MessagingExtensionResponse> OnTeamsMessagingExtensionQueryAsync(ITurnContext<IInvokeActivity> turnContext, MessagingExtensionQuery action, CancellationToken cancellationToken)
+        {
```

Comment:
it seems this file is not required, could you please verify once? most of the methods are implemented in DialogBot.cs file, please remove this file if not required